src/Serializers/Native.php

```diff
@@ -460,6 +460,12 @@ protected function mapByReference(&$data)
 
             $instance = $data;
 
+            if ($data instanceof \DateTime) {
```

Comment:
I don't think this is enough. I think we need to check instanceof the DateTimeInterface, so that immutable objects are handled correctly, too